src/background/index.ts

```diff
@@ -0,0 +1,76 @@
+import '@/lib/initPolyfills';
+import { PopupActionType } from '@/types';
+import { openPopup, removePopup, getPopup } from './bgPopupHandler';
+import updateDynamicRules from './redirectRule';
+
+let creating: Promise<void> | null; // A global promise to avoid concurrency issues
```

Comment:
If the function `setupOffscreenDocument` is fired only once right after its definition how this is possible that the `creating` value could have different values?

Reply:
Correct